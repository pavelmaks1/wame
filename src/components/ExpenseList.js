import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
  <div>
    <h1>Expense list</h1>
    {console.log(props)}
    {props.expenses.map( (expense,index) =>
      <ExpenseListItem 
        {...expense} 
        key={expense.id}
        indexId={index}
      />
    )}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

console.log(mapStateToProps);

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList);

export default ConnectedExpenseList;