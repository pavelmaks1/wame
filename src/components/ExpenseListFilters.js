import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate , endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  render () {
    return  (
      <div>
        <input 
          type="text" 
          value={this.props.filters.text} 
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select value={this.props.filters.sortBy} onChange={ (e) => {
          if (e.target.value === 'date') {
            this.props.dispatch(sortByDate());
          } else if ( e.target.value === 'amount') {
            this.props.dispatch(sortByAmount());
          }
        }} >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
          {console.log('that', this.props.filters)}
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          startDateId="MyDatePicker"
          endDate={this.props.filters.endDate}
          endDateId="MyDatePicker2"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          showClearDates={true}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    )
  }
}



const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);