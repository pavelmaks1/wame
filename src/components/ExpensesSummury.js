import React from 'react';
import { getExpensesTotal } from '../selectors/expenses-total';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummury = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? 'Expense' : 'Expenses';
  const formattedExpenseTotal = numeral(expenseTotal / 100).format('$0,0.00');

  return (
    <div>
      <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal} </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: getExpensesTotal(visibleExpenses)
  }

};

export default connect(mapStateToProps)(ExpensesSummury);