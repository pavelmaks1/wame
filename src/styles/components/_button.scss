.button {
  background: $blue;
  color: #fff;
  display: inline-block;
  border: none;
  font-weight: 300;
  padding: $s-size;
  line-height: 1;
  font-size: $font-size-large;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #c51111;
}