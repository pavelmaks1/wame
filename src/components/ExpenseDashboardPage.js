import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummury from './ExpensesSummury';
import { LoginPage } from './LoginPage';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
    <ExpensesSummury />
  </div>
);

export default ExpenseDashboardPage;
