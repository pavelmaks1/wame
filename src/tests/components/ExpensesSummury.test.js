import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummury } from '../../components/ExpensesSummury';

test('should correctly render ExpensesSummury with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummury expenseCount={1} expenseTotal={15} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummury with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummury expenseCount={20} expenseTotal={200} />);
  expect(wrapper).toMatchSnapshot();
});