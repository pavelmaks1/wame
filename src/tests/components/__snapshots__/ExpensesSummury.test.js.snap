// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummury with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    Expense
     totalling 
    $0.15
     
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummury with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    20
     
    Expenses
     totalling 
    $2.00
     
  </h1>
</div>
`;
